src/cogent3/app/tree.py

```diff
@@ -127,15 +116,11 @@ def __init__(self, drop_invalid=False):
             calculated are excluded, the resulting tree will be for the subset of labels with strictly valid distances
             if False, an ArithmeticError is raised if a distance could not be computed on observed data.
         """
-        super(quick_tree, self).__init__(
-            input_types=self._input_types,
-            output_types=self._output_types,
-            data_types=self._data_types,
-        )
-        self._formatted_params()
         self._drop_invalid = drop_invalid
 
-    def main(self, dists):
+    T = Union[SerialisableType, AlignedSeqsType]
+
+    def main(self, dists: T) -> Union[SerialisableType, TreeType]:
```

Comment:
change to `dists: PairwiseDistanceType`

Reply:
Fist I had written `PairwiseDistanceType` for this. But for a test error, I changed it to `Union[SerialisableType, AlignedSeqsType]`.

Reply:
That seems problematic. In that case, better to leave that app un decorated so we can tackle it separately.

Reply:
I'd rather see such errors than just mark as `xfail`.